### DISCUSSION TITLE
Question about io.to(arr)

### DISCUSSION THREAD

**Discussion by SupertigerDev** [socketio/socket.io#4378] [Q&A]
When I use the code below, it only emits to the users that are joined to the 2 rooms.
```js
io.to(["room1", "room2"]).emit(...)
```` 
However, when the array is empty, it is emitting to everyone. I don't know if this is a bug or what but I dont want it to emit to anyone if the array is empty.
```js
io.to([]).emit(...)
```

**Comment by darrachequesne** [comment]
Hi! There is indeed currently no distinction between emitting empty array (`io.to([]).emit(...)`) and no array (`io.emit(...)`):

- https://github.com/socketio/socket.io/blob/5ab8289c0acd33d7230541f4fe3be5ccca21a556/lib/broadcast-operator.ts#L17
- https://github.com/socketio/socket.io-adapter/blob/b92d65cf9c6124930fac78349e2a0b847fe19f16/lib/index.ts#L290

We could add one though. What do you think?

**Reply by SupertigerDev** [reply]
Hello! Yes that would be amazing ❤️
